operational/azure/azure_long_running_instances/azure_long_running_instances.pt

```diff
@@ -0,0 +1,257 @@
+    end
+  end
+end
+
+# Lists All Virtual Machines in each subscription
+datasource "ds_vm_list" do
+  # https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/list-all?tabs=HTTP
+  iterate $ds_subscription_list
+  request do
+    auth $auth_azure
+    pagination $pagination_azure
+    host "management.azure.com"
+    path join(["/subscriptions/", val(iter_item,"subscriptionId"), "/providers/Microsoft.Compute/virtualMachines"])
+    query "api-version","2022-08-01"
+    query "statusOnly", "true"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", jmes_path(col_item,"id")
+      field "rg", get(4, split(jmes_path(col_item,"id"), "/"))
+      field "name", jmes_path(col_item,"name")
+      field "statuses", jmes_path(col_item,"properties.instanceView.statuses")
+      field "subscriptionId",val(iter_item,"subscriptionId")
+      field "subscriptionName",val(iter_item,"displayName")
+    end
+  end
+end
+
+# Calls the script to get only the virtual machines that are running
+datasource "ds_vm_running" do
+    run_script $js_filter_vm_running, $ds_vm_list, $param_days_old
+  end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+# Returns instances that are long running
+script "js_filter_vm_running", type: "javascript" do
+  parameters "ds_vm_list", "param_days_old"
+  result "result"
+  code <<-EOS
+  var result=[];
+  _.each(ds_vm_list, function(instance){
+    var codes = _.pluck(instance.statuses, 'code')
+    if ( _.contains(codes, "PowerState/running") ) {
+      var runningState = _.where(instance.statuses, {"code": "PowerState/running"});
+      var displayStatus = runningState[0]["displayStatus"];
+      var provisionState = _.where(instance.statuses, {"code": "ProvisioningState/succeeded"});
+      var timeRunning = provisionState[0]["time"];
+      var timeofevent = new Date(timeRunning);
+      var nowtime = Date.now();
+      var res = Math.abs(timeofevent.valueOf() - nowtime.valueOf());
+      var daysElapsed = Math.ceil(res / (1000 * 3600 * 24));
+      result.push({
+        "name": instance["name"],
+        "status": displayStatus,
+        "time_running": timeRunning,
+        "days_elapsed": daysElapsed,
+        "resourcegroup" : instance["rg"]
+        "subscriptionId" : instance["subscriptionId"]
+        "subscriptionName" : instance["subscriptionName"]
+      })
+    }
+  })
+  EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_utilization" do
+  validate_each $ds_vm_running do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Azure VMs in Stopped State for Over {{parameters.param_days_old}} Day(s)"
```

Comment:
this works but I'd suggest changing it to what we expect in policy incident format, which is through the "export" keyword